client/src/App.tsx

```diff
@@ -0,0 +1,59 @@
+import { Routes, Route, Navigate, useNavigate } from 'react-router-dom';
+import { NavBar, PageType } from './NavBar';
+import { InsideWebsiteNavBar, PageTypeInsideApp } from './InsideWebsiteNavBar';
+import { IconClicked } from './IconClicked';
+import RegistrationForm from './RegistrationForm';
+import SignInForm from './SignInForm';
+import FeedComponent from './FeedComponent';
+import RecommendationComponent from './RecommendationComponent';
+import RatingComponent from './RatingComponent';
+import Profile from './Profile';
+import './App.css';
+
+export default function App() {
+  const navigate = useNavigate();
+  const isAuthenticated = !!sessionStorage.getItem('token');
+
+  function handleNavigate(page: PageType) {
+    navigate(page);
+    if (page === 'sign-out') {
+      sessionStorage.removeItem('token');
+      navigate('sign-in');
+    }
+  }
+
+  function handleNavigateInsideApp(pageNew: PageTypeInsideApp) {
+    navigate(pageNew);
+    if (pageNew === 'Logout') {
+      sessionStorage.removeItem('token');
+      navigate('Logout');
```

Comment:
Since pageNew is 'Logout', at line 26 you already navigated to 'Logout', so this is redundant.

Simpler would be to do the `if` first and remove the token, then navigate always.